config/stack/manifests/resources/bucketclass.ui-schema.yaml

```diff
@@ -0,0 +1,5 @@
+version: 0.4
+configSections: []
+
+resourceConditionPaths:
+- .status.conditionedStatus.conditions[?(@.type=='Ready')]
```

Comment:
class types don't have any status as far as i know, they are basically just a template.  i don't think having a condition path in class types makes sense.

Reply:
In response to https://github.com/crossplaneio/stack-gcp/pull/63#pullrequestreview-307305438 and this comment:

* Since there is currently no consumer of these annotations, the format of the annotations can only be derived and verified from the design document.  It is not clear how error conditions, like missing fields, would be handled.  The Stack Manager unit tests in the Crossplane repo verify that these files would be picked up and concatenated to the ui-schema annotations of the CRD.  Since there is no consumer, it is not clear if potential consumers would have benefitted from a single (group level) ui-schema.yaml.
* `Classes` do not expose the Ready status condition (they have no direct controller), so a group level ui-schema.yaml would not make sense in this case, I'll remove `resourceConditionPaths` values from the `Class` types, but I'll leave the files in place for future updates.
  I `git grep`'d for `runtimev1alpha1.Available` to find that the controller backed types generally supported this status:

  ```console
  $ git grep runtimev1alpha1.Available | grep -v -e _test.go -e yaml 
  pkg/clients/connection/connection.go:                           s.SetConditions(runtimev1alpha1.Available())
  pkg/clients/globaladdress/globaladdress.go:             s.SetConditions(runtimev1alpha1.Available())
  pkg/controller/cache/managed.go:                cr.Status.SetConditions(runtimev1alpha1.Available())
  pkg/controller/compute/gke.go:  instance.Status.SetConditions(runtimev1alpha1.Available(), runtimev1alpha1.ReconcileSuccess())
  pkg/controller/compute/network.go:      cr.Status.SetConditions(runtimev1alpha1.Available())
  pkg/controller/compute/subnetwork.go:   cr.Status.SetConditions(runtimev1alpha1.Available())
  pkg/controller/storage/bucket.go:       bh.setStatusConditions(runtimev1alpha1.Available(), runtimev1alpha1.ReconcileSuccess())
  ```
  
  I'll do a little more digging.
* The shared managed reconciler pattern provides this field and setters for it.  `cr.Status.SetConditions(runtimev1alpha1.Available())`. `Ready` status is available in all of resources that take advantage of it (which I'm told is all of the GCP, Azure, AWS managed resources).  `status.bindingPhase` is also available in resources where binding makes sense.

Reply:
I've pushed updates for the GCP, Azure, AWS, and Rook stacks. I counted the ducks this time.

I also included a small fix for the example install.stack.yaml which was using an invalid metadata.name in Azure and AWS.  I modified it to use a namespace and name patterned after the GCP and Rook stacks.

Reply:
These are the only resources that are including the `Ready` status now:

```
stack-aws/.../dbsubnetgroup
stack-aws/.../ekscluster
stack-aws/.../iamrolepolicyattachment
stack-aws/.../iamrole
stack-aws/.../internetgateway
stack-aws/.../rdsinstance
stack-aws/.../replicationgroup
stack-aws/.../routetable
stack-aws/.../s3bucket
stack-aws/.../securitygroup
stack-aws/.../subnet
stack-aws/.../vpc
stack-azure/.../account
stack-azure/.../akscluster
stack-azure/.../container
stack-azure/.../mysqlserver
stack-azure/.../mysqlservervirtualnetworkrule
stack-azure/.../postgresqlserver
stack-azure/.../postgresqlservervirtualnetworkrule
stack-azure/.../redis
stack-azure/.../resourcegroup
stack-azure/.../subnet
stack-azure/.../virtualnetwork
stack-gcp/.../bucket
stack-gcp/.../cloudmemorystoreinstance
stack-gcp/.../cloudsqlinstance
stack-gcp/.../connection
stack-gcp/.../gkecluster
stack-gcp/.../globaladdress
stack-gcp/.../network
stack-gcp/.../subnetwork
stack-rook/.../cockroachcluster
stack-rook/.../yugabytecluster
```